src/lib.rs

```diff
@@ -221,8 +245,8 @@ impl<C> Rcan<C> {
         self.payload.capability()
     }
 
-    pub fn capability_key(&self) -> &VerifyingKey {
-        self.payload.capability_key()
+    pub fn capability_origin(&self) -> &CapabilityOrigin {
+        self.payload.capability_origin()
```

Comment:
Why not both?

```suggestion
    pub fn capability_key(&self) -> &VerifyingKey {
        match self.payload.capability_origin {
            CapabilityOrigin::Issuer => &self.issuer,
            CapabilityOrigin::Delegation(ref from) => from,
        }
    }

    pub fn capability_origin(&self) -> &CapabilityOrigin {
        &self.payload.capability_origin
```

Also, avoid endless recursion ;)